runtime-sdk/src/types/address.rs

```diff
@@ -291,6 +298,30 @@ impl From<Address> for ConsensusAddress {
     }
 }
 
+/// Generate a custom Ethereum address with proper domain separation.
+pub fn generate_custom_eth_address(module: &str, kind: &[u8]) -> [u8; 20] {
```

Comment:
Maybe `module` is a bit misleading since it's not necessarily an oasis-sdk module name passed in.